src/CoinbaseSdk/Core/credentials/CoinbaseCredentials.cs

```diff
@@ -16,86 +16,46 @@
 
 namespace CoinbaseSdk.Core.Credentials
 {
-  using System;
-  using System.Security.Cryptography;
-  using System.Text;
-  using CoinbaseSdk.Core.Error;
+    using System;
+    using System.Security.Cryptography;
+    using System.Text;
+    using CoinbaseSdk.Core.Error;
 
-  /// <summary>
-  /// Class that represents the credentials used to authenticate with the Coinbase API.
-  /// </summary>
-  public class CoinbaseCredentials
-  {
     /// <summary>
-    /// Initializes a new instance of the <see cref="CoinbaseCredentials"/> class.
+    /// Class that represents the credentials used to authenticate with the Coinbase API.
     /// </summary>
-    /// <param name="accessKey">Coinbase API AccessKey.</param>
-    /// <param name="passphrase">Coinbase API Secret Key.</param>
-    /// <param name="signingKey">Coinbase API Signing Key.</param>
-    public CoinbaseCredentials(
-      string accessKey = null,
-      string passphrase = null,
-      string signingKey = null)
+    public class CoinbaseCredentials(string accessKey, string passphrase, string signingKey)
```

Comment:
this will not error if they pass in null as any of these values. you want to only allow the public to be get, and use an exception in the constructor to set

public class CoinbaseCredentials
{
    public string AccessKey { get; }
    public string Passphrase { get; }
    public string SigningKey { get; }

    public CoinbaseCredentials(string accessKey, string passphrase, string signingKey)
    {
        AccessKey = accessKey ?? throw new ArgumentNullException(nameof(accessKey));
        Passphrase = passphrase ?? throw new ArgumentNullException(nameof(passphrase));
        SigningKey = signingKey ?? throw new ArgumentNullException(nameof(signingKey));
    }
}